scripts/camera.lua

```diff
@@ -292,4 +310,59 @@ function Camera.setTransitionPeriod(camera, transitionPeriod)
     Camera.updateConfig(camera)
 end
 
+--- @param player LuaPlayer
+--- @param camera Camera.camera
+function Camera.updateChartTags(player, camera)
+    -- we can't do this without a player or force
+    if not player or not player.force then
+        return
+    end
+    
+    -- we have duplicated calculations for recovering tile coordinates from center/zoom
+    -- if we had a third box, I think it would be worth extracting this logic/data somehow
+
+    --- do the work of deleting and reinstating the tag
+    local function modifyTag(name, pos, icon)
```

Comment:
`name` is only used to store/fetch the tags. The `icon.name` is also suitable for this and saves a function parameter

Reply:
this is true of the code as committed - it was not true in the past

Reply:
This suggestion is implemented in the new `refreshChartTags` method.